Automattic/jetpack
# Show the correct homepage when previewing a theme in the Site Editor

## Proposed changes:

Show the correct homepage when previewing a theme in the Site Editor to support [Block Theme Previews](https://github.com/WordPress/gutenberg/pull/50030) on dotcom. This is done by overriding the `show_on_front` option to `posts`.

I added changes in ~~Jetpack~~ jetpack-mu-wpcom since dotcom wants to support Block Theme Previews both on Simple and Atomic sites. 

This PR is part of https://github.com/Automattic/wp-calypso/issues/79221.

### Other information:


## Jetpack product discussion




## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






Atomic sites

* Create an Atomic site
* Change your Reading Settings to show a static page as a homepage
* Patch this PR to your site by following https://github.com/Automattic/jetpack/pull/31929#issuecomment-1639638285
	* Note that you must set `define( 'JETPACK_MU_WPCOM_LOAD_VIA_BETA_PLUGIN', true );` to your `wp-config.php`
* Install a theme 
* Go to Block Theme Previews with `/wp-admin/site-editor.php?wp_theme_preview=<your-theme>` (e.g. `?wp_theme_preview=pendant`)
* Verify Block Theme Previews show Index or Home template as a homepage

Simple sites

- Create a site
- Change your Reading Settings to show a static page as a homepage
- Patch this PR to your sandbox by following https://github.com/Automattic/jetpack/pull/31929#issuecomment-1639638285
- Sandbox your testing site AND `public-api.wordpress.com`
- Go to Block Theme Previews with `/wp-admin/site-editor.php?wp_theme_preview=<your-theme>` (e.g. `?wp_theme_preview=pub/pendant`)
- Verify Block Theme Previews show Index or Home template as a homepage

self-hosted site

- Patch this PR to your site
	- I used `jetpack rsync` PCYsg-eg0-p2
- Go to Block Theme Previews with `/wp-admin/site-editor.php?wp_theme_preview=<your-installed-theme>` (e.g. `?wp_theme_preview=twentytwentytwo`)
- Verify nothing breaks on the Site Editor
